pymathics/natlang/linguistic_data.py

```diff
@@ -0,0 +1,364 @@
+        elif isinstance(word, String) or word.get_head() is SymbolList:
+            pass
+        else:
+            return
+
+        wordnet, language_code = self._load_wordnet(
+            evaluation, self._language_name(evaluation, options)
+        )
+        if not wordnet:
+            return
+
+        py_word = self._parse_word(word)
+        if not py_word:
+            return
+
+        senses = self._senses(py_word, wordnet, language_code)
+        if senses is not None:
+            return ListExpression(*[[String(s) for s in desc] for syn, desc in senses])
+
+    def eval_property(self, word, property, evaluation: Evaluation, options: dict):
+        "WordData[word_, property_String, OptionsPattern[WordData]]"
+        if word.get_head is SymbolStringExpression:
+            if property.get_string_value() == "Lookup":
+                return Expression(SymbolDictionaryLookup, word)
+        elif isinstance(word, String) or word.get_head() is SymbolList:
+            return self._property(
+                word, property.get_string_value(), "ShortRules", evaluation, options
+            )
+
+    def eval_property_form(
+        self, word, property, form, evaluation: Evaluation, options: dict
+    ):
+        "WordData[word_, property_String, form_String, OptionsPattern[WordData]]"
+        if isinstance(word, String) or word.get_head() is SymbolList:
+            return self._property(
+                word,
+                property.value,
+                form.value,
+                evaluation,
+                options,
+            )
+
+
+class WordDefinition(_WordNetBuiltin):
+    """
+    <url>:WMA link:
+    https://reference.wolfram.com/language/ref/WordDefinition.html</url>
+
+    <dl>
+      <dt>'WordDefinition[$word$]'
+      <dd>returns a definition of $word$ or Missing["Available"] if $word$ is not known.
+    </dl>
+
+    >> WordDefinition["gram"]
+     = {a metric unit of weight equal to one thousandth of a kilogram}
+    """
+
+    summary_text = "retrieve the definition of a word"
+
+    def eval(self, word, evaluation: Evaluation, options: dict):
+        "WordDefinition[word_String, OptionsPattern[WordDefinition]]"
+        wordnet, language_code = self._load_wordnet(
+            evaluation, self._language_name(evaluation, options)
+        )
+        if wordnet:
+            senses = self._senses(word.value.lower(), wordnet, language_code)
+            if senses:
+                return ListExpression(*[String(syn.definition()) for syn, _ in senses])
+            else:
+                return Expression(SymbolMissing, StringNotAvailable)
+
+
+class WordList(_WordListBuiltin):
+    """
+    <url>:WMA link:
+    https://reference.wolfram.com/language/ref/WordList.html</url>
+
+    <dl>
+      <dt>'WordList[]'
+      <dd>returns a list of common words.
+
+      <dt>'WordList[$type$]'
+      <dd>returns a list of common words of type $type$.
+    </dl>
+
+    >> Length[WordList[]] > 10000
```

Comment:
The test check if this is a list with at least 10000 elements. I did this in this way to avoid a failure due to differences in the length of the dictionary across versions.

Reply:
I moved this test to pymathics, and used a text similar to the case with the second parameter here.